game-core/src/test/java/games/strategy/triplea/UnitUtilsTest.java

```diff
@@ -191,7 +191,9 @@ void transportedUnitsAreTransferredToTheFirstNewUnit() {
       gameData.performChange(changes);
 
       assertThat(
-          "The first new unit should be the new transporter for the units",
+          "Units can only be transported by one unit at a time. So the transported unit "
```

Comment:
I get the sense that the updated description is missing the mark on explaining why the expectation is what it is. 

When reading this method:
- see we have an 'oldUnit', that is actually an 'oldTransport' and we place in a SZ
- we then createa  'newUnit' that is actually a 'newTransport' and we do not place it in a SZ
- we then create an infantry in the same SZ and add them to 'oldTransport'
- we then perform the change action
- now we assert that the transporter for the old unit is now the new transport


Recommendations:
1. Keep the unit types explicit and avoid the list operations. The list access makes it harder to grok. Instead of 'transportedUnits.get(0)', it's easier to understand if written as 'oldInfantry'. EG:
```
final Unit oldTransport = ...
final Unit infantryOnTransport = ...

final Unit newTransport = ...

 final Change changes = UnitUtils.translateAttributesToOtherUnits(List.of(oldTransport),  List.of(newTransport);
```

Then:
```
assertThat(
   "Transport swapped from old to new, the transported infantry on the old transport should now be on the new transport",
  infantryOnTransport.getTransportedBy(), is(newTransport));
```

Let me know if that clarifies how the test cases can be made more clear. After looking at that, I'm curious if the inverted change will fix the above. I also wonder as well what happens if a transported infantry is part of the change list and is replaced by a new infantry. Does the old transported infantry get assigned to the new transport? (IE: what happens if we had a 'newInfantry' in the mix above, would both 'oldInfantry' and 'newInfantry' be marked as transported by the new transport?)

Reply:
Thinking about the above further, I think it does speak to more test cases. If we have more than one infantry on transport, all infantry are assigned to the new transport? (which means in theory one could have looped over the list of transported units and asserted they all had the new transport. Given it's a singleton list, I see why you did not, though explicitly checking the first list element makes it seem overtly intentional. Though, I do think keeping the units separate and not in the list makes it easier to follow along and avoids the "okay, we are grabbing the first unit off this list, which unit is that?" (Which is a salient question for 'newUnits.get(0)', it's not clear from that line that would be the transport)

Reply:
@DanVanAtta , I've update this test in https://github.com/triplea-game/triplea/pull/8414/commits/234d8ac9f2aae3d19ff90c4c9c77a030fd4e9d48.  It does look you misread it.  This is testing the case where a single transporter is converted into two transporters.  The unit being transported (the infantry) needs to go to one of the new transporters.  It doesn't make sense for the infantry to be in both of the transports and the code doesn't allow it either.

As for the question of what happens if a transported unit gets changed?  I don't know.  Currently, they are always killed.  Reason for that is that when a transported unit is killed, it is done after units are transformed.  And a transported unit currently can't be damaged.  We'd have to modify this code to translate the `transportedBy` of the `unitGivingAttributes` to the `unitsThatWillGetAttributes`.

Reply:
I also added tests for multiple transported/unloaded items in https://github.com/triplea-game/triplea/pull/8414/commits/5ec7f757a31c71b53e8ea2b3cbc29ee36c4637ff.